### PR TITLE
Initial implementation of Transformer.

### PR DESCRIPTION
**Pull request by marcushanses**
A first iteration of the Transformer class, it enables users to traverse the tree built with isly.

### CODE REVIEW

File: Transformer.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import { tuple } from "./tuple"
+import { array, optional, readonly as islyReadonly, Type } from "./Type"
+import { islyUndefined as undefined } from "./undefined"
+import { union } from "./union"
+import { unknown } from "./unknown"
+
+export abstract class Transformer<T> {
+	transform(type: Transformer.Types, options?: Transformer.Options): T | undefined {
+		let result: T | undefined
+		type instanceof array.Class && type.class
```
Comment (AntonCroner) [inline Transformer.ts:21]: useless

**Review by AntonCroner — CHANGES_REQUESTED** [review]
Seems fine. We should probably bump a minor.

File: Transformer.ts (hunk 1 above)
Comment (marcushanses) [reply Transformer.ts]: You are quite correct!

**Review by AntonCroner — APPROVED** [review]

**Comment by tvartom** [comment]
Update Readme?

**Pull request merged**